test/tests/functional/pbs_sched_preempt_enforce_resumption.py

```diff
@@ -105,10 +105,9 @@ def test_suspended_job_ded_time_calendared(self):
         end = start + 120
         self.scheduler.add_dedicated_time(start=start, end=end)
 
-        temp += 180
         j1 = Job(TEST_USER)
         j1.set_attributes({ATTR_l + '.select': '1:ncpus=2',
-                           ATTR_l + '.walltime': temp})
+                           ATTR_l + '.walltime': 180})
```

Comment:
I think this test will fail when on line 103 value of temp is evaluated to be 60. That would mean dedicated time will start 4 minutes in the future. low priority job will run for 3 mins. High priority job will preempt the low priority job for 60 seconds and then low priority job will resume back for the remainder of the time. 

Line 103 is causing the problem in this test. I don't think the test needs the complication of doing "now % 60" to start a dedicated time in the future. 
I'd say just, keep your current changes and remove line 103, update start time to something like "now +240" and that's it.

Reply:
Dedicated time starts on the minute boundary.  That's what the now % 60 is for.  It's to floor it to the minute boundary.

Reply:
Oh ok, I remember now, dedicated time does not take in seconds. In that case, how about increasing the walltime of J1 by a second and make it 181.

Reply:
If we want a job that will conflict with dedicated time, I'd say 200 or more.

Reply:
hmmm... that may not work because the job may not even start if there is a dedicated time starting in less than 200 seconds. How about instead of giving it a hardcoded value, just make walltime of the job as "temp -5" which will mean the job will end 5 seconds earlier than the dedicated time. It will also result in conflict when this job is suspended and resumed after 60 seconds.

Reply:
Hi @arungrover , if we use temp-5 according to me it wont work because temp value in between 0-60 and if we use it then job J1 will suspend and J2 will resume but J2 is only 60 sec. and dedicated start time will start at least after 180 sec. so J1 will resume before start_time and test will fail at line 133.

if we put walltime value less than 180 sec. like 170 sec. or 175 sec. then as mentioned by you if we get temp value 0 then also dedicated start_time will start after 180 sec. from current time and J1 will comes in R state. test will not fail at line 133

Reply:
ya, for it to work you will have to retain line 108 that you removed in this PR.